core-bundle/src/Translation/Translator.php

```diff
@@ -76,7 +76,7 @@ public function getLocale(): string
         return $this->translator->getLocale();
     }
 
-    public function getCatalogue($locale = null): MessageCatalogue
+    public function getCatalogue($locale = null): MessageCatalogueInterface
```

Comment:
That change is unrelated, I think?

Reply:
No, that's intentional and correct this way in terms of LSP (,MHO). Otherwise mocking is impossible.

Reply:
The return type was changed from `MessageCatalogue` to `MessageCatalogueInterface` in https://github.com/contao/contao/pull/3276 
But yes, the return type as defined in the `TranslatorBagInterface` is indeed supposed to be `MessageCatalogueInterface`.

/cc @ausi

Reply:
We always return an instance of `Contao\CoreBundle\Translation\MessageCatalogue` so IMO this was correct.
I’d mock the translations by setting `$GLOBALS['TL_LANG'][…]`. 
But we can also change the `MessageCatalogue` from `final` to `@final`.
But I’m also fine with the change to `MessageCatalogueInterface`, so it’s your choice ☺️

Reply:
> We always return an instance of Contao\CoreBundle\Translation\MessageCatalogue so IMO this was correct.

Yes, it is correct from what the current implementation is in the method, but, I think, when I want to work with a instance of `TranslatorBagInterface`, I only ever want/need to get a `MessageCatalogueInterface` (that's what the contract defined by the interface says). Our `MessageCatalogue` does not add anything that is not covered by the interface. So I only see downsides in not using the interface as the return type. :smile: 

Does that make sense? If so, I'd go with the current version. :relaxed:

Reply:
Also if you decorate `Contao\CoreBundle\Translation\MessageCatalogue` this would also fail, no?